### ISSUE TITLE
Users who are mutual blocked hosted sessions still apear in world browser.

### ISSUE THREAD

**Issue by ExoTheWicker** [Yellow-Dog-Man/Resonite-Issues#1643]
### Describe the bug?

I have a user mutual blocked and I can still see sessions they're hosting in the world browser.

### To Reproduce

Block someone who is hosting a public session and that world still apears.appears

### Expected behavior

Sessions hosted by users you have blocked should not appear in world browser.

### Resonite Version Number

Beta 2024.4.3.1170

### What Platforms does this occur on?

Windows

### What headset if any do you use?

Index

### Log Files

[DESKTOP-I5BSEJ5 - 2024.4.3.1170 - 2024-04-07 23_09_49.log](https://github.com/Yellow-Dog-Man/Resonite-Issues/files/14900427/DESKTOP-I5BSEJ5.-.2024.4.3.1170.-.2024-04-07.23_09_49.log)

### Reporters

ExoTheWicker

**Comment by JackTheFoxOtter** [comment]
I would say this is probably intended? To me it doesn't make much sense to hide sessions based on if the host is blocked or not.

What would make more sense to me would be showing an information hint in the session browser when a blocked user is in that session.

**Comment by Frooxius** [comment]
It's definitely not a bug - we didn't intend to hide the sessions.

The main question here is - should these sessions actually be hidden? Even if you have mutual block, there might be other people in that session that you might want to see.

While you can't join the session since you have blocked the host of it, it might still be useful to know where are other friends at, rather than removing this information.

**Comment by TisFoolish** [comment]
In my experience if a person blocks someone online the service will generally hide all details related to that person. As an example on Reddit if you were to block someone all their comments their posts even their profile is completely hidden from you. 

And if your friends are in a session that you've blocked the host of then the game could just say "*Blocked session of X User*" instead of the world name that way the user still knows that their friends are all together and why they won't be able to join.

**Comment by Frooxius** [comment]
It depends on what kind of block it is.

With this, you're blocking the person (not a session), so you don't see them when you're in worlds with them, but I don't think that necessarily implies that you want to block out all the information where are people at.

If the information is reduced to just "Session of a blocked user", then you're losing more context on where your friends are at - what world they're in, type of session, easily viewable list of users and so on. I don't think it's given that user want to block out all that information when blocking a person.

**Comment by TisFoolish** [comment]
I personally disagree, but at the very least there should be something denoting that you've blocked the host

**Comment by Frooxius** [comment]
I think generally the best answer for this is configuration, so you can determine what you want to see and what not.

The current blocking functionality as it stands is made to be quick to cover most pressing use-cases. Because of that I don't want to make too many assumptions for the simplified system.

**Comment by Draconas** [comment]
I have a close enough problem that I don't think I would need to make another GH issue.
>I was friends with "bob"
>bob blocked me
>I removed him from my contacts, however bob didn't remove me
>as such, bob will show in the "contacts session" portion of the world browser
I want this to not be a thing anymore, being advertised a session I'm not allowed in honestly blows

**Comment by shiftyscales** [comment]
@Draconas - Receiving contact updates from a user you are not a contact of sounds like a bug, but is outside of the scope of this request.

It looks like Easton opened an issue about that one here: https://github.com/Yellow-Dog-Man/Resonite-Issues/issues/2031

**Comment by SlyTheFloof** [comment]
yea

**Comment by alexderpyfox** [comment]
finding your friends in a world of someone you blocked is horrible you rather not know where they are

**Comment by Zefnoly** [comment]
> finding your friends in a world of someone you blocked is horrible you rather not know where they are

I totally agree... The problem with many blocking systems on some platforms is being forced to see stuff related to said person over and over. Blocking someone does mean "do not want to see or interact" with said person anymore. That should mutually include sessions they host. May as well rename the feature to "do not interact" rather than "block" otherwise. One thing ive learned about blocking features that dont fully block is that blocking becomes almost pointless and can lead to further distress...

**Comment by Zefnoly** [comment]
> I would say this is probably intended? To me it doesn't make much sense to hide sessions based on if the host is blocked or not.
> 
> What would make more sense to me would be showing an information hint in the session browser when a blocked user is in that session.

It does not make much sense to call it a blocking feature if you are stil forced to visually interact or display said person and their stuff. Then it essentially just becomes a "do not interact" or "ignore" feature instead..

**Comment by SlyTheFloof** [comment]
> > I would say this is probably intended? To me it doesn't make much sense to hide sessions based on if the host is blocked or not.
> > What would make more sense to me would be showing an information hint in the session browser when a blocked user is in that session.
> 
> It does not make much sense to call it a blocking feature if you are stil forced to visually interact or display said person and their stuff. Then it essentially just becomes a "do not interact" or "ignore" feature instead..

i 100% agree with that i can see everyone still cus thay don't kill the nameplates properly and it's been a issue with #1385 anyway of knowing a blocked user exists in a session will cause some sort of issue users that are also blocked are still visable on the userinspector if you dig for it as a value user override it seems aswell as you still can see them in the user session tab and still jump to them if you so wanted

**Comment by alexderpyfox** [comment]
It makes me sad now because I had to block people that used to be friends but still seeing their world appear in the worlds list and seeing other people in their world via the contacts list is a stab in the chest 😞😞😞😞 I don't want to be reminded of it I just want to move on

**Comment by SlyTheFloof** [comment]
> It makes me sad now because I had to block people that used to be friends but still seeing their world appear in the worlds list and seeing other people in their world via the contacts list is a stab in the chest 😞😞😞😞 I don't want to be reminded of it I just want to move on

10000% agree

**Comment by Dusty-Sprinkles** [comment]
There are definitely users who when I block them, I block them because of the sessions they host/may host, so I would love to have this where I can fully hide blocked users' hosted sessions, I think I care less about sessions with blocked non-host users in them though.
